### ISSUE TITLE
Secure archiver service endpoints

### ISSUE THREAD

**Issue by aaclan-ebi** [ebi-ait/hca-ebi-dev-team#184]
Dev notes:
- Archiver service can be an internal service.
- Ingest Core API can wrap the archiver service endpoints.

**Comment by aaclan-ebi** [comment]
Archiver deployment doc https://docs.google.com/document/d/1G5EufiGT3gJhf-Z3xDbTtAUmYDcsBcXzc0VymSG6AIs/edit#heading=h.3m2wwe8dp3cc

**Comment by aaclan-ebi** [comment]
Archiver deployment doc https://docs.google.com/document/d/1G5EufiGT3gJhf-Z3xDbTtAUmYDcsBcXzc0VymSG6AIs/edit#heading=h.3m2wwe8dp3cc

**Comment by clairerye** [comment]
Just double checking @aaclan-ebi this ticket is to secure the API https://api.ingest.archive.data.humancellatlas.org/browser/index.html#/

**Comment by aaclan-ebi** [comment]
Hi @clairerye ! Nope this is different. The archiver service is exposed here https://archiver.ingest.dev.archive.data.humancellatlas.org/ . I think we can just make it internal and wrap behind https://api.ingest.archive.data.humancellatlas.org so we can easily secure all Ingest API endpoints.

**Comment by clairerye** [comment]
So do we need a separate ticket for securing https://api.ingest.archive.data.humancellatlas.org?

**Comment by aaclan-ebi** [comment]
Yes, i think this ticket is specific archiver endpoints. Some changes needed are for reconfiguring the archiver deployment, and implementing wrapper endpoints in Ingest Core. That's much more work needed than just securing the current PUT/DELETE/GET endpoints in Ingest Core which is I think you want to focus on for next sprint? 

I'll create a new one now.

**Comment by aaclan-ebi** [comment]
Putting this back to product backlog for now as this is not part of current sprint goal.

**Comment by aaclan-ebi** [comment]
Putting back to sprint backlog based from @clairerye's suggestion to @prabhat to work on this during standup. (Not sure which is which with ebi-ait/hca-ebi-dev-team#194 )

**Comment by prabh-t** [comment]
It was decided that we secure the archiver service endpoints using api key instead of the initial approach of wrapping endpoints in core and making the archiver an internal service only. This was due to a number of reasons including added complexity/duplication in core.
The api key auth is now implemented in the following PRs
https://github.com/ebi-ait/ingest-archiver/pull/20
https://github.com/ebi-ait/ingest-kube-deployment/pull/8
